### PR TITLE
UserSettings: Document userArgs

### PR DESCRIPTION
**Pull request by bgamari**

### CODE REVIEW

File: src/UserSettings.hs [hunk 1]
```diff
@@ -3,8 +3,8 @@
 -- If you don't copy the file your changes will be tracked by git and you can
 -- accidentally commit them.
 module UserSettings (
-    userBuildRoot, userFlavours, userPackages, verboseCommand,
-    buildProgressColour, successColour, stage1Only
+    userBuildRoot, userFlavours, userPackages, userArgs,
```
Comment (snowleopard) [inline src/UserSettings.hs]: Why is `userArgs` exported?

File: src/UserSettings.hs [hunk 2]
```diff
@@ -36,6 +36,10 @@ userFlavour = defaultFlavour { name = "user" } -- Modify other settings here.
 userPackages :: [Package]
 userPackages = []
 
+-- | Arguments to pass to builders. See @doc/user-settings.md@.
```
Comment (snowleopard) [inline src/UserSettings.hs]: Presumably, you meant to use `userArgs` when defining `userFlavour`?

See example here: https://github.com/snowleopard/hadrian/blob/master/doc/user-settings.md#command-line-arguments

**Comment by snowleopard** [comment]
@bgamari Thanks, we do need to make `UserSettings` easier to use. We don't need to export `userArgs`, we need to use it in this line:
```haskell
userFlavour = defaultFlavour { name = "user" } -- Modify other settings here.
```

**Comment by bgamari** [comment]
Indeed I completely misunderstood the documentation here. Ignore me,

**Comment by snowleopard** [comment]
I refuse to ignore you -- I'll try to improve the docs! :)

**Comment by bgamari** [comment]
Thanks @snowleopard!

**Pull request closed without merging**